### PR TITLE
New global and interactive plotting

### PR DESCRIPTION
**Pull request by jhennawi**
This PR implements two new features: 

1) The interactive plotting with the -s option that allows the code to continue running. I would appreciate it if you guys could give this a spin in the test suite and provide some feedback. I would like to know if there are bugs on Linux etc.

You might consider changing the backend to

backend :WebAgg

In your .matplotlib/matplotlibrc

This has the effect of sending all the interactive plots to a web browser via the WebAgg server. This is nice because a) the plots can all be toggled through via web-browser tags, but don't pop up all over your screen b) the plots are all interactive while the code runs, c) it gets around the annoying bug with MacOSX that causes multiprocessing guis to crash. 

If we like this WebAgg behavior we could consider setting it in pypeit when we load matplotlib. 

The only slightly annoying thing is that when the plots launch, the browser takes the mouse focus from whatever window you were in, but this may be a platform dependent thing, and there is probably a way to toggle it off. The easy way to deal with it is to just make the web browser window small. 

2) I've re-worked the global sky subtraction to now also do polynomial 2-d fitting with the b-spline. This results in dramatically improved global fits. Not sure why were not doing this in low-redux. It slightly increases the time for the global sky-subtraction. We need to test this thouroughly, because the increased complexity of the fits may also make fits less stable sometimes. I noticed that DEIMOS is going through many iterations before the fits finally stabilize and converge, so we may need to increase the b-spline spacing to stabilize the matrices. 

Joe

### CODE REVIEW

File: pypeit/core/extract.py [hunk 1]
```diff
@@ -481,9 +484,14 @@ def findfwhm(model, sig_x):
 def qa_fit_profile(x_tot,y_tot, model_tot, l_limit = None, r_limit = None, ind = None,
                    title =' ', xtrunc = 1e6, xlim = None, ylim = None, qafile = None):
 
+    from importlib import reload
```
Comment (profxj) [inline pypeit/core/extract.py]: remove this and all calls of reload

File: pypeit/core/extract.py [hunk 2]
```diff
@@ -1078,12 +1106,21 @@ def fit_profile(image, ivar, waveimg, trace_in, wave, flux, fluxivar,
...
+         #              l_limit = l_limit, r_limit = r_limit, ind = ss[inside], xlim = prof_nsigma)
+
+        p_show_profile = multiprocessing.Process(target=qa_fit_profile, args=(sigma_x, norm_obj/(pb + (pb == 0.0)), full_bsp),
+                                 kwargs={'l_limit': l_limit, 'r_limit': r_limit, 'title': title_string, 'ind': ss[inside],
+                                         'xlim': prof_nsigma})
+        p_show_profile.daemon = True
+        p_show_profile.start()
 
-    return (profile_model, xnew, fwhmfit, med_sn2)
+    return (profile_model, xnew, fwhmfit, med_sn2, p_show_profile)
```
Comment (profxj) [inline pypeit/core/extract.py]: no () needed.

File: pypeit/core/extract.py [hunk 3]
```diff
@@ -1657,566 +1721,20 @@ def objfind(image, thismask, slit_left, slit_righ, inmask = None, FWHM = 3.0,
...
-        for i in range(ny):
-            # Weight the Vandermonde matrix for this y vector
-            _vand = w[i,:,None] * vand
-            c = np.linalg.lstsq(_vand, __y[i,:])[0]
-            ym[i,:] = np.sum(c[:,None] * vand.T[:,:], axis=0)
-
-    # Return the model with the appropriate shape
-    return ym if y.ndim == 2 else ym[0,:]
 
+    return (sobjs, skymask[thismask], objmask[thismask], proc_list)
```
Comment (profxj) [inline pypeit/core/extract.py]: no ()

File: pypeit/core/load.py [hunk 4]
```diff
@@ -336,17 +336,19 @@ def load_1dspec(fname, exten=None, extract='OPT', objname=None, flux=False):
     rsp_kwargs['wave_tag'] = '{:s}_WAVE'.format(extract)
     if flux:
         rsp_kwargs['flux_tag'] = '{:s}_flam'.format(extract)
-        rsp_kwargs['var_tag'] = '{:s}_flam_var'.format(extract)
+        rsp_kwargs['ivar_tag'] = '{:s}_flam_ivar'.format(extract)
```
Comment (profxj) [inline pypeit/core/load.py]: let's CAPITALIZE these

**Comment by profxj** [comment]
I ran through with `-s` and there were new plots in my Ginga
but nothing else.  Here was my precise call:

`run_pypeit shane_kast_red_600_7500_d55.pypeit -v 2 -o -s`

File: pypeit/core/extract.py (hunk 1 above)
Comment (jhennawi) [reply pypeit/core/extract.py]: Sorry this was a debugging thing.

**Comment by profxj** [comment]
@jhennawi -- Here is my backend:

# If you omit this parameter, it will always default to "Agg", which is a
# non-interactive backend.
backend      : qt5agg

File: pypeit/core/load.py (hunk 4 above)
Comment (jhennawi) [reply pypeit/core/load.py]: Done.

File: pypeit/core/extract.py (hunk 3 above)
Comment (jhennawi) [reply pypeit/core/extract.py]: done.

**Pull request merged**